src/basic/fs-util.c

```diff
@@ -197,16 +197,18 @@ int readlink_and_make_absolute(const char *p, char **r) {
 int chmod_and_chown_at(int dir_fd, const char *path, mode_t mode, uid_t uid, gid_t gid) {
         _cleanup_close_ int fd = -1;
 
-        assert(dir_fd >= 0 || dir_fd == AT_FDCWD);
+        assert(dir_fd >= 0 || (dir_fd == AT_FDCWD && path));
 
         if (path) {
                 /* Let's acquire an O_PATH fd, as precaution to change mode/owner on the same file */
                 fd = openat(dir_fd, path, O_PATH|O_CLOEXEC|O_NOFOLLOW);
                 if (fd < 0)
                         return -errno;
+
+                dir_fd = fd;
         }
```

Comment:
maybe instead add another branch here:

```c
if (path) {
        …
else if (dir_fd == AT_FDCWD) {
        /* Neither a dir_fd nor a path specified? then operate on the cwd */
        fd = openat(dir_fd, ".", O_PATH|O_CLOEXEC|O_NOFOLLOW|O_DIRECTORY);
        if (fd < 0)
                return -errno;
        dir_fd = fd;
}
```

or in other words: just make this work? AT_FDCWD after all means "operate on current working dir", so we better should